### ISSUE TITLE
Buffer is not 'modifiable'

### ISSUE THREAD

**Issue by bugsbugsbux** [hrsh7th/nvim-cmp#1113]
### FAQ

- [X] I have checked the [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt) and it didn't resolve my problem.

### Announcement

- [X] I have checked [Breaking change announcement](https://github.com/hrsh7th/nvim-cmp/issues/231).

### Minimal reproducible full config

```
if has('vim_starting')
  set encoding=utf-8
endif
scriptencoding utf-8

if &compatible
  set nocompatible
endif

let s:plug_dir = expand('/tmp/plugged/vim-plug')
if !filereadable(s:plug_dir .. '/plug.vim')
  execute printf('!curl -fLo %s/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim', s:plug_dir)
end

execute 'set runtimepath+=' . s:plug_dir
call plug#begin(s:plug_dir)
Plug 'hrsh7th/nvim-cmp'
Plug 'hrsh7th/cmp-buffer'
Plug 'hrsh7th/cmp-nvim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'neovim/nvim-lspconfig'
call plug#end()
PlugInstall | quit

" Setup global configuration. More on configuration below.
lua << EOF
local cmp = require "cmp"
cmp.setup {
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },

  mapping = {
    ['<CR>'] = cmp.mapping.confirm({ select = true })
  },

  sources = {
    { name = "nvim_lsp" },
    { name = "buffer" },
  },
}

cmp.setup.cmdline(':', {
    mapping = cmp.mapping.preset.cmdline({}),
    sources = cmp.config.sources({
        {name = 'nvim_lsp'},
        {name = 'buffer'},
    }),
})
vim.cmd[[autocmd FileType markdown nnoremap <buffer> <F5> <cmd>pedit +set\ nomodifiable <cfile><cr>]]
EOF

lua << EOF
local capabilities = require('cmp_nvim_lsp').update_capabilities(vim.lsp.protocol.make_client_capabilities())

require'lspconfig'.cssls.setup {
  capabilities = capabilities,
}
EOF
```

### Description

Im developing a wiki plugin which opens markdown links in the preview window. Once this happened completion in the commandline breaks (see error below)...

### Steps to reproduce

$ echo hello world > preview.md
$ echo '[preview](preview.md)' > index.md
$ nvim index.md
type (i mean the key f5; not angle, capital f...):
`10l<F5>:p`

### Expected behavior

completion works without error

### Actual behavior

```
Error executing vim.schedule lua callback: ...d/vim-plug/nvim-cmp/lua/cmp/view/custom_entries_view.lua:142: Buffer is not 'modifiable'
stack traceback:
        [C]: in function 'nvim_buf_set_lines'
        ...d/vim-plug/nvim-cmp/lua/cmp/view/custom_entries_view.lua:142: in function 'open'
        /tmp/plugged/vim-plug/nvim-cmp/lua/cmp/view.lua:110: in function 'open'
        /tmp/plugged/vim-plug/nvim-cmp/lua/cmp/core.lua:323: in function 'fn'
        /tmp/plugged/vim-plug/nvim-cmp/lua/cmp/utils/async.lua:41: in function </tmp/plugged/vim-plug/nvim-cmp/lua/cmp/utils/async.lua:39>
```

**Comment by hrsh7th** [comment]
Perfect reproduce step. But I can't understand what's the cause of this... I'll investigate it.

**Comment by listout** [comment]
Facing a similar issue with man pages. When I try to search using `/` inside a manpage I get the same error message.

Also possible duplicate of [this](https://github.com/hrsh7th/cmp-buffer/issues/37) issue from the cmp-buffer.

**Comment by jimmyjansen93** [comment]
Having the same issue here on manpages as I have `cmp` enabled for commands. As a temporary workaround, I use the following snippet:
```
vim.api.nvim_create_autocmd({ "FileType" }, {
  pattern = { "help", "man", "lspinfo", "spectre_panel" },
  callback = function()
    vim.cmd [[
      nnoremap <silent> <buffer> q :q!<CR> 
      set nobuflisted 
    ]]
  end,
})
```
With the above, I can easily quit with `q`, this does not resolve the `/` problem, though. 

Whenever I have time, I will implement a workaround that enables/disables `cmp` based on whether the editor is modifiable. 

@hrsh7th perhaps that is also the solution for this problem?

**Comment by marcusatiliusregulus** [comment]
This is strange: If I call `:pedit | pc` **at least once before** I first open a non-modifiable preview buffer, everything **works fine**...

**Comment by alyson-ferreira** [comment]
I got the same error doing the following (using [lsp-zero](https://github.com/VonHeikemen/lsp-zero.nvim) plugin default configuration for nvim-cmp):

1. Open a lua file (or any other file) `nvim ~/.config/nvim/init.lua`.
2. Don't interact with the file (to avoid creating buffer for cmp hints I believe).
3. Open another file in a new window `:vsplit ~/.zshrc`.
4. Make it not modifiable `:set noma` (now any new buffer will be read-only).
5. Go back to init.lua `<CTRL_W> W`.
6. Edit it and cause completion view to show (Ctrl-e is the default in lsp-zero). The error will show up.

I was able to **avoid** the error by using `:setlocal noma` on step 3, instead.

Hope it helps somehow.
